File: RecyclerViews/RecyclerViewSwipeRefresh/app/src/main/java/edu/cs4730/recyclerviewswiperefresh/myAdapter.java
package edu.cs4730.recyclerviewswiperefresh;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import edu.cs4730.recyclerviewswiperefresh.databinding.MyRowBinding;

/**
 * this adapter is very similar to the adapters used for listview, except a ViewHolder is required
 * see http://developer.android.com/training/improving-layouts/smooth-scrolling.html
 * except instead having to implement a ViewHolder, it is implemented within
 * the adapter.
 */

public class myAdapter extends RecyclerView.Adapter<myAdapter.ViewHolder> {

    private String[] values;
    private List<String> myList;
    private Context mContext;
    private Random mRandom = new Random();

    public myAdapter(String[] values, Context context) {
        this.values = values;
        this.mContext = context;
        //now get actual list to display.
        randomlist();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        MyRowBinding binding = MyRowBinding.inflate(LayoutInflater.from(mContext), viewGroup, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int i) {
        String entry = myList.get(i);
        viewHolder.viewBinding.Name.setText(entry);
        viewHolder.viewBinding.Name.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TextView tv = (TextView) v;
                Toast.makeText(mContext, tv.getText(), Toast.LENGTH_SHORT).show();
            }
        });
        viewHolder.viewBinding.picture.setImageResource(R.drawable.phone);
    }

    @Override
    public int getItemCount() {
        return myList == null ? 0 : myList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        MyRowBinding viewBinding;

        public ViewHolder(MyRowBinding viewBinding) {
            super(viewBinding.getRoot());
            this.viewBinding = viewBinding;
        }
    }

    //a simple method to create a random list of vlaues.
    public void randomlist() {
        myList = new ArrayList<String>();
        for (int i = 0; i < 10; i++) {
            myList.add(values[(mRandom.nextInt(values.length - 1))]);
        }
    }

    //a simple method to remove an item from the list for swipe left.
    public void removeitem(int i) {
        myList.remove(i);
        notifyDataSetChanged();
    }
}

File: RecyclerViews/RecyclerViewSwipeRefresh_kt/app/src/main/java/edu/cs4730/recyclerviewswiperefresh_kt/myAdapter.kt
package edu.cs4730.recyclerviewswiperefresh_kt

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import edu.cs4730.recyclerviewswiperefresh_kt.databinding.MyRowBinding
import java.util.*

/**
 * this adapter is very similar to the adapters used for listview, except a ViewHolder is required
 * see http://developer.android.com/training/improving-layouts/smooth-scrolling.html
 * except instead having to implement a ViewHolder, it is implemented within
 * the adapter.
 */
class myAdapter(
    private val values: Array<String>,  private val mContext: Context
) : RecyclerView.Adapter<myAdapter.ViewHolder>() {
    private var myList: MutableList<String>? = null
    private val mRandom = Random()
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val v = MyRowBinding.inflate(LayoutInflater.from(mContext), viewGroup, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val entry = myList!![i]
        viewHolder.viewBinding.Name.text = entry
        viewHolder.viewBinding.Name.setOnClickListener { v ->
            val tv = v as TextView
            Toast.makeText(mContext, tv.text, Toast.LENGTH_SHORT).show()
        }
        viewHolder.viewBinding.picture.setImageResource(R.drawable.phone)
    }

    override fun getItemCount(): Int {
        return if (myList == null) 0 else myList!!.size
    }
    inner class ViewHolder(var viewBinding: MyRowBinding) : RecyclerView.ViewHolder(viewBinding.root) {
    }

    //a simple method to create a random list of vlaues.
    fun randomlist() {
        myList = ArrayList()
        for (i in 0..9) {
            (myList as ArrayList<String>).add(values[mRandom.nextInt(values.size - 1)])
        }
    }

    //a simple method to remove an item from the list for swipe left.
    fun removeitem(i: Int) {
        myList!!.removeAt(i)
        notifyDataSetChanged()
    }

    init {
        //now get actual list to display.
        randomlist()
    }
}